app/scripts/store.js

```diff
@@ -220,6 +220,22 @@ var exports = (function() {
       });
     },
 
+    resetModel: function (modelContainer) {
+      return new Promise((resolve, reject) => {
+        if (modelContainer === undefined || modelContainer.id === undefined) {
```

Comment:
We wouldn't, but I want this method to fail elegantly by rejecting the promise with a message if for some reason this occurs.